sqlx-mysql/src/options/mod.rs

```diff
@@ -414,6 +504,24 @@ impl MySqlConnectOptions {
         self.set_names = flag_val;
         self
     }
+
+    /// Sets the compression mode for the connection.
+    ///
+    /// Data is uncompressed by default.
+    /// Ensure that the server supports the selected compression algorithm;
+    /// if it does not, the client will fall back to uncompressed mode.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use sqlx_mysql::{MySqlConnectOptions, Compression};
+    /// let options = MySqlConnectOptions::new()
+    ///     .compression(Compression::Zlib.fast());
+    /// ```
+    pub fn compression(mut self, compression: CompressionConfig) -> Self {
```

Comment:
It's probably important to specify that the compression level setting only takes effect on the server for Zstd.

For zlib compression, this only affects outgoing packets; the server hardcodes a compression level of 6: https://github.com/mysql/mysql-server/blob/056a391cdc1af9b17b5415aee243483d1bac532d/sql/auth/sql_authentication.cc#L3238